corehq/apps/app_manager/suite_xml/features/case_tiles.py

```diff
@@ -165,3 +185,21 @@ def _case_tile_template_string(self):
         """
         with open(case_tile_template_config(self.detail.case_tile_template).filepath, encoding='utf-8') as f:
             return f.read()
+
+    def _get_xpath_mapped_to_field_containing_sort(self):
+        xpath_to_field = {}
+        for column_info in self.detail_column_infos:
+            # column_info is an instance of DetailColumnInfo named tuple. It has the following properties:
+            #   column_info.column: an instance of app_manager.models.DetailColumn
+            #   column_info.sort_element: an instance of app_manager.models.SortElement
+            #   column_info.order: an integer
+            from corehq.apps.app_manager.detail_screen import get_column_generator
+            fields = get_column_generator(
+                self.app, self.module, self.detail,
+                detail_type=self.detail_type, *column_info
+            ).fields
+            for field in fields:
+                if field.sort_node:
+                    xpath_func = self._get_xpath_function(column_info.column)
+                    xpath_to_field[xpath_func] = field
```

Comment:
Good point - I also don't expect there to be a use case where the same column would be configured with two different sorts but it would be good to protect against it.
[1d96b11](https://github.com/dimagi/commcare-hq/pull/33125/commits/1d96b115633476c50764ade00f34488dff0fd720)